enrollment-server-onboarding-provider-innovatrics/src/main/java/com/wultra/app/onboardingserver/provider/innovatrics/InnovatricsDocumentVerificationProvider.java

```diff
@@ -55,31 +119,250 @@ public boolean shouldStoreSelfie() {
+     * @throws DocumentVerificationException if the returned data could not be provided.
+     */
+    private String getExtractedData(final String customerId, final OwnerId ownerId) throws RemoteCommunicationException, DocumentVerificationException {
+        return serializeToString(innovatricsApiService.getCustomer(customerId, ownerId));
+    }
+
+    /**
+     * Checks if a document portrait of the customer is available.
+     * @param customerId id of the customer.
+     * @param ownerId owner identification.
+     * @return true if document portrait is available, false otherwise.
+     */
+    private boolean hasDocumentPortrait(final String customerId, final OwnerId ownerId) throws RemoteCommunicationException {
+        return innovatricsApiService.getDocumentPortrait(customerId, ownerId).isPresent();
+    }
+
+    /**
+     * Creates DocumentSubmitResult containing extracted data.
+     * @param customerId id of the customer to get data from.
+     * @return DocumentSubmitResult containing extracted data.
+     */
+    private static DocumentSubmitResult createSubmitResult(final String customerId, final SubmittedDocument submitted) {
+        final DocumentSubmitResult result = new DocumentSubmitResult();
+        result.setUploadId(customerId);
+        result.setDocumentId(submitted.getDocumentId());
+        result.setExtractedData(DocumentSubmitResult.NO_DATA_EXTRACTED);
+        return result;
+    }
+
+    /**
+     *
+     * @param customerId id of the customer the document belongs to.
+     * @param ownerId owner identification.
+     * @return DocumentVerificationResult
+     */
+    private DocumentVerificationResult createVerificationResult(String customerId, OwnerId ownerId) throws DocumentVerificationException, RemoteCommunicationException {
+        final DocumentInspectResponse response = innovatricsApiService.inspectDocument(customerId, ownerId);
+
+        final List<String> rejectionReasons = new ArrayList<>();
+        if (Boolean.TRUE.equals(response.getExpired())) {
+            rejectionReasons.add("Document expired.");
+        }
+
+        if (response.getMrzInspection() != null && !Boolean.TRUE.equals(response.getMrzInspection().getValid())) {
+            rejectionReasons.add("MRZ does not conform the ICAO specification.");
+        }
+
+        final VisualZoneInspection viz = response.getVisualZoneInspection();
+        if (viz != null) {
+            if (!CollectionUtils.isEmpty(viz.getOcrConfidence().getLowOcrConfidenceTexts())) {
+                rejectionReasons.add("Low OCR confidence of text.");
+            }
+            if (viz.getTextConsistency() != null && !Boolean.TRUE.equals(viz.getTextConsistency().getConsistent())) {
+                rejectionReasons.add("Inconsistent text field.");
+            }
+        }
+
+        if (response.getPageTampering() != null) {
+            response.getPageTampering().forEach((k, v) -> {
```

Comment:
Can we avoid using one letter variable names? This is more C-style variable naming.